velox/type/Filter.h

```diff
@@ -261,6 +310,10 @@ class AlwaysFalse final : public Filter {
  public:
   AlwaysFalse() : Filter(true, false, FilterKind::kAlwaysFalse) {}
 
+  folly::dynamic serialize() const override;
+
+  static FilterPtr create(const folly::dynamic& /*obj*/);
```

Comment:
We should change `deserialize` here to return `unique_ptr` then.  `unique_ptr` can be moved into `shared_ptr` implicitly so we probably do not need to change the use sites.

Reply:
The change should not be too large, you should only need to change the registry and existing deserialization implementations.  No need to change TypePtr because there is implicit conversion from unique_ptr<T> to shared_ptr<T>.  You can do the registry and other deserializer part in a separate PR.

Reply:
OK, got it. Add a new registry (`DeserializationRegistryForUniquePtr`) and update the Filters' create methods.
Will refactor the serialization framework using unique_ptr in a separate PR.